### ISSUE TITLE
False positive ts(1100) - Invalid use of 'arguments' in strict mode.

### ISSUE THREAD

**Issue by burtek** [microsoft/TypeScript#50063]
# Bug Report

### 🔎 Search Terms

invalid use arguments strict mode

### 🕗 Version & Regression Information

Existed in `3.3.3`, exists in `4.7.4`, `4.8.0-beta` and `Nightly`.

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `arguments`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.8.0-dev.20220726#code/MYewdgzgLgBAhjAvDA3vATgcwK4FsCmYUEAXDANowCMANDAEx0DMMAujAL4CwAULwGbYwwKAEtwMfgAoAlKl4xFMAPTKY+dOhDoyASTAA3OABtRAExjYI+GCH4wA5HCx5CxBzFFgY0dKJEwuCBm+AB0MFJUVAAM0TIKSqCQsGjOOAREEJxI8ADcCYro+FDY6N5prpn5PNx8PEnQMJg5skgAfPI8SipqGlo6MPpGphZWNnaOFRnunt6+-rBBIeGRMXEFMA0pGOluWRw5cNXdRSVlO5XE1bW8WzAAFi1yiB0oG6rqmtp6hibmltZbPYnC5phAPF4fFA-AElmEIlFYvEuolwI0pntDqEMZkNqdSuVQXteLUgA)

### 💻 Code

```ts
const a = { arguments: [ 1, 2, 3 ] }

function f() {
    // error: Invalid use of 'arguments' in strict mode. (1100)
    const { arguments } = a;
    return arguments;
}

const g = () => {
    // error: Invalid use of 'arguments' in strict mode. (1100)
    const { arguments } = a;
    return arguments;
}

const h = () => {
    // error: Invalid use of 'arguments' in strict mode. (1100)
    const arguments = a.arguments
    return arguments
}

```

### 🙁 Actual behavior

Error is thrown upon using `arguments` as variable name, both in destructuring assignment and in a "normal" variable declaration.

### 🙂 Expected behavior

No error should be thrown as the `arguments` name is not reference to [`arguments` object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments).

**Comment by MartinJohns** [comment]
`arguments` is not a legal identifier in strict-mode.

**Comment by burtek** [comment]
@MartinJohns oh, I thought it's just illegal to reference the `arguments` object, not that the identifier is illegal. My bad

**Comment by Josh-Cena** [comment]
Btw, `arguments` as a call-scoped variable is not deprecated. Only `someFunction.arguments` as a property and the `arguments.callee` property is.
